docbot/action_test.go

```diff
@@ -0,0 +1,326 @@
+				ID:     &id,
+				Body:   &body,
+				Labels: nil,
+			},
+		), mock.WithRequestMatch(
+			mock.GetReposLabelsByOwnerByRepo,
+			repoLabels(),
+		),
+		mock.WithRequestMatch(mock.PostReposIssuesLabelsByOwnerByRepoByIssueNumber, nil),
+		mock.WithRequestMatch(mock.GetReposIssuesCommentsByOwnerByRepoByIssueNumber, nil),
+		mock.WithRequestMatch(mock.PostReposIssuesCommentsByOwnerByRepoByIssueNumber, nil),
+	)
+
+	config := mustNewActionConfig()
+	action := NewActionWithClient(context.Background(), config, github.NewClient(mockedHTTPClient))
+
+	err := action.Run(1, "opened")
+	assertMessageLabel(t, err, MessageLabelMultiple)
+}
+
+func TestSingleChecked_WhenLabelMissingExist(t *testing.T) {
+	id := int64(1)
+	body := fmt.Sprintf(`
+Check the box below or label this PR directly.
+
+Need to update docs?
+
+- [ ] %s
+(Your PR needs to update docs and you will update later)
+
+- [x] %s
+(Please explain why)
+
+- [ ] %s
+(Your PR contains doc changes)
+
+- [ ] %s
+(Docs have been already added)
+`, "`doc-required`", "`doc-not-needed`", "`doc`", "`doc-complete`")
+
+	labelMissing := "doc-label-missing"
+	mockedHTTPClient := mock.NewMockedHTTPClient(
+		mock.WithRequestMatch(
+			mock.GetReposPullsByOwnerByRepoByPullNumber,
+			github.PullRequest{
+				ID:     &id,
+				Body:   &body,
+				Labels: []*github.Label{{Name: &labelMissing}},
+			},
+		), mock.WithRequestMatch(
+			mock.GetReposLabelsByOwnerByRepo,
+			repoLabels(),
+		),
+		mock.WithRequestMatch(mock.PostReposIssuesLabelsByOwnerByRepoByIssueNumber, nil),
+		mock.WithRequestMatch(mock.DeleteReposIssuesLabelsByOwnerByRepoByIssueNumberByName, nil),
+	)
+
+	config := mustNewActionConfig()
+	action := NewActionWithClient(context.Background(), config, github.NewClient(mockedHTTPClient))
+
+	err := action.Run(1, "opened")
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestUnchecked_WhenLabelMissingExist(t *testing.T) {
+	id := int64(1)
+	body := fmt.Sprintf(`
+Check the box below or label this PR directly.
+
+Need to update docs?
+
+- [ ] %s
+(Your PR needs to update docs and you will update later)
+
+- [ ] %s
+(Please explain why)
+
+- [ ] %s
+(Your PR contains doc changes)
+
+- [ ] %s
+(Docs have been already added)
+`, "`doc-required`", "`doc-not-needed`", "`doc`", "`doc-complete`")
+
+	labelMissing := "doc-label-missing"
+	mockedHTTPClient := mock.NewMockedHTTPClient(
+		mock.WithRequestMatch(
+			mock.GetReposPullsByOwnerByRepoByPullNumber,
+			github.PullRequest{
+				ID:     &id,
+				Body:   &body,
+				Labels: []*github.Label{{Name: &labelMissing}},
+			},
+		), mock.WithRequestMatch(
+			mock.GetReposLabelsByOwnerByRepo,
+			repoLabels(),
+		),
+	)
+
+	config := mustNewActionConfig()
+	action := NewActionWithClient(context.Background(), config, github.NewClient(mockedHTTPClient))
+
+	err := action.Run(1, "opened")
+	assertMessageLabel(t, err, MessageLabelMissing)
+}
+
+func TestName(t *testing.T) {
+	split := strings.Split("docs, docs-required,", ",")
+	t.Log(split)
+}
```

Comment:
This test should test the functionality of `NewActionConfig` directly with assertions.

Reply:
Removed